### ISSUE TITLE
bbox的loss计算

### ISSUE THREAD

**Issue by turboLIU** [AITTSMD/MTCNN-Tensorflow#78]
代码中，bbox是（batchsize，4）的，应该是每个batch中只有一个bbox把，，但是实际情况也不只有一个目标，，一副图像中多个bbox的loss是怎么计算的，，代码没看懂，求大神解释一下
bbox_target = tf.placeholder(tf.float32, shape=[config.BATCH_SIZE, 4], name='bbox_target')

**Comment by guojiapeng00** [comment]
1个batch中有好多bbox呀，（每个正样本都有一个）
bbox——loss代码：（4个坐标值求差，再平方，最后求均值）
def bbox_ohem(bbox_pred,bbox_target,lbel):
    zeros_index = tf.zeros_like(label, dtype=tf.float32)
    ones_index = tf.ones_like(label,dtype=tf.float32)
    valid_inds = tf.where(tf.equal(tf.abs(label), 1),ones_index,zeros_index)
    #(batch,)
    square_error = tf.square(bbox_pred-bbox_target)
    square_error = tf.reduce_sum(square_error,axis=1)
    #keep_num scalar
    num_valid = tf.reduce_sum(valid_inds)
    #keep_num = tf.cast(num_valid*num_keep_radio,dtype=tf.int32)
    keep_num = tf.cast(num_valid, dtype=tf.int32)
    #keep valid index square_error
    square_error = square_error*valid_inds
    _, k_index = tf.nn.top_k(square_error, k=keep_num)
    square_error = tf.gather(square_error, k_index)
    return tf.reduce_mean(square_error)

**Comment by turboLIU** [comment]
@guojiapeng00 应该是一个batch有多个bbox，，但是代码中的bbox_target的shape是（batchsize，4），
如果是多个bbox，那placeholder的shape应该是（batchsize，4*n）吧，4的倍数！

**Comment by guojiapeng00** [comment]
你指的n，应该是batchsize

**Comment by guojiapeng00** [comment]
1张图片----（1,4）
batch_size张-------(batch_size,4)

**Comment by turboLIU** [comment]
如果一张图片有10个目标，那应该是（1，40）吧

**Comment by guojiapeng00** [comment]
训练时，正样本只有一个人脸

**Comment by turboLIU** [comment]
@guojiapeng00 为啥？不是应该一张图片的所有信息都丢进去训练吗？难道训练时，一张图片只取一个正样本？其他正样本下次迭代再计算？

**Comment by Hzzone** [comment]
@turboLIU 构建图像金字塔，正、部分、负样本，你应该再看一下数据处理方面的代码
